### PR TITLE
Update installation page 

### PR DESCRIPTION
**Pull request by potiuk**
The installation page is updated with:

* updated Python/Kubernetes support policies
* added download information to installation page

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Small but I think worth merging before 2.1 for the download links

**Comment by potiuk** [comment]
@kaxil @ashb - small one to add downloads before 2.1

File: docs/exts/extra_files_with_substitutions.py [hunk 1]
```diff
@@ -25,19 +25,28 @@ def copy_docker_compose(app, exception):
...
-                    output.write(line.replace('|version|', app.config.version))
+                    output_file.write(line.replace('|version|', app.config.version))
+
+    # Replace `|version|` in the installation.html that requires manual substitutions (in links)
+    for path in app.config.manual_substitutions_in_generated_html:
+        with open(os.path.join(app.outdir, os.path.basename(path))) as input_file:
+            content = input_file.readlines()
+        with open(os.path.join(app.outdir, os.path.basename(path)), "wt") as output_file:
+            for line in content:
+                output_file.write(line.replace('|version|', app.config.version))
```
Comment (ashb) [inline docs/exts/extra_files_with_substitutions.py:41]: This is coming from install.rst right? If so then Sphinx's native replacement should already handle this!

Comment (potiuk) [reply docs/exts/extra_files_with_substitutions.py:41]: Not really - it does not handle it when they are part of URL :(

Comment (potiuk) [reply docs/exts/extra_files_with_substitutions.py:41]: `Apache Airflow |version| sdist package <https://downloads.apache.org/airflow/|version|/apache-airflow-|version|-bin.tar.gz>`_ (`asc <https://downloads.apache.org/airflow/|version|/apache-airflow-|version|-bin.tar.gz.asc>`__, `sha512 <https://downloads.apache.org/airflow/|version|/apache-airflow-|version|-bin.tar.gz.sha512>`__)

**Review by ashb — APPROVED** [review]
May be a better fix we could do for `|version|` in links, but this'll do for now

**Pull request merged**